src/main/clientHandler/ClientHandlerThread.java

```diff
@@ -0,0 +1,49 @@
+package clientHandler;
+
+import messageExtractor.JSONExtractor;
+import messageExtractor.PeripheralAction;
+import messageExtractor.UnexpectedActionException;
+import clientInteractors.ClientInteractorInterface;
+import peripherals.ActionNotDefinedException;
+import peripherals.IPeripheral;
+import peripherals.Lock;
+
+import java.io.IOException;
+
+public class ClientHandlerThread implements Runnable{
+    private ClientInteractorInterface clientInteractor;
+
+    public ClientHandlerThread(ClientInteractorInterface clientIntractor) {
+        this.clientInteractor = clientIntractor;
+    }
+
+    @Override
+    public void run() {
```

Comment:
In our current architecture, we create a new thread for each message the server receives. Should we change this?